codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/config/ClientConfig.java

```diff
@@ -181,7 +181,6 @@ public Builder name(String name) {
          */
         public Builder type(String type) {
```

Comment:
After seeing the changes - I would recommend:
- change this method to validateType and have it add a TypeConstraint (instead of needing to do `.constraint(new TypeConstraint)`).  
- remove `type` from the builder.  We shouldn't have it - its confusing.  There is no "type" for config, theres multiple different contexts for type.
- Remove the inference logic in the constructor and require that documetnationType AND rbsType are specified.  

this should create a consistent, explicit, easy to read/use builder.